processors/hra_linker/processor.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+# Module API
+
+def process(conf, conn):
+
+    # Iterate over all hra publications
+    count = 0
+    for publication in base.readers.read_rows(conn, 'database', 'publications',
+            orderby='id', source_id='hra'):
+
+        # Find identifiers
+        identifiers = base.helpers.find_identifiers(
+            publication['title'] + publication['abstract'])
+
+        # Delete existent relationships
+        conn['database']['trials_publications'].delete(
+            publication_id=publication['id'])
+
+        # Write new relationships
+        for identifier in identifiers:
+
+            # Get trial
+            trial_facts = base.helpers.slugify_array([identifier])
+            trial = base.readers.read_objects(
+                conn, 'trials', first=True, facts=trial_facts)
+
+            # Found trial - add relationship
+            if trial:
+                base.writers.write_trial_relationship(
+                    conn, 'publication', publication, publication['id'], trial['id'])
+                logger.debug('Linked %s to "%s"',
+                    trial['primary_id'], publication['title'][0:50])
+
+            # Not found trial - add trial stub
+            else:
+                trial = {
+                    'primary_register': 'Health Research Authority',
+                    'primary_id': identifier,
+                    'identifiers': {
+                        'hra': identifier,
+                    },
+                    'public_title': identifier,
+                }
+                base.writers.write_trial(conn, trial, 'hra')
```

Comment:
I imagine `base.writers.write_trial` will add the trials' `primary_source_id` to `hra`, right?